napari/utils/key_bindings.py

```diff
@@ -182,6 +187,257 @@ def normalize_key_combo(key_combo):
+        """
+        if name not in self._actions:
+            return
+        buttons = self._buttons[name]
+        desc = self._actions[name].description
+
+        # update buttons with shortcut and description
+        if name in self._shortcuts:
+            sht = self._shortcuts[name]
+            sht_str = f' ({format_shortcut(sht)})'
+        else:
+            sht = ''
+            sht_str = ''
+        for button in buttons:
+            button.setToolTip(desc + sht_str)
+        if name in self._qactions:
+            action = self._actions[name]
+            qaction = self._qactions[name]
+            qaction.setShortcut(
+                QKeySequence(sht.replace('-', '+').replace('Control', 'Ctrl'))
+            )
+            qaction.setText(action.description)
+            qaction.setStatusTip(action.description)
+
+    def _update_shortcut_bindings(self, name):
+        """
+        Update the key mappable for given action name
+        to trigger the action within the given context and
+        make all the corresponding UI element flash if possible.
+        """
+        if name not in self._actions:
+            return
+        action = self._actions[name]
+        if name not in self._shortcuts:
+            return
+        sht = self._shortcuts.get(name)
+        keymappable = action.keymappable
+        if hasattr(keymappable, 'bind_key'):
+
+            def flash(*args):
+                call_with_context(action.command, self.context)
+                for b in self._buttons.get(name, []):
+                    b._animation.start()
+
+            keymappable.bind_key(sht)(flash)
+
+    def bind_button(self, name, button):
+        """
+        Bind `button` to trigger Action `name` on click.
+        """
+        action = self._actions[name]
+
+        button.clicked.connect(
+            lambda: call_with_context(action.command, self.context)
+        )
+
+        self._buttons[name].append(button)
+        self._update_gui_elements(name)
+
+    def bind_shortcut(self, name, shortcut):
+        """
+        bind shortcut `shortcut` to trigger action `name`
+        """
+        self._shortcuts[name] = shortcut
+        self._update_shortcut_bindings(name)
+        self._update_gui_elements(name)
+
+    def bind_qaction(self, name, qaction):
+        """
+        Bind the given qaction to an action.
+
+        This will also update the description and shortcut when those changes.
+
+        Same as for shortcut; make ui element flash when action is triggered.
+        """
+        self._qactions[name] = qaction
+        action = self._actions[name]
+
+        def flash(*args):
+            call_with_context(action.command, self.context)
+            for b in self._buttons.get(name, []):
+                b._animation.start()
+
+        qaction.triggered.connect(flash)
+        self._update_gui_elements(name)
+
+    def unbind_shortcut(self, name):
```

Comment:
This assumes one shortcut per action, which might be the current case, but moving forward there might be cases where 2 shortcuts per action may be desirable, in that case we would need to pass the actual shortcut we are unbinding?